### PR TITLE
Add Ruby 3.2, 3.3, HEAD to CI matrix

### PR DESCRIPTION
**Pull request by s4na**

### CODE REVIEW

File: ci/sunspot_test_script.sh [hunk 1]
```diff
@@ -13,7 +13,13 @@ start_solr_server() {
   # go to sunspot_solr folder and install dependencies
   current_path=`pwd`
   cd ../sunspot_solr
-  bundle install --quiet --path vendor/bundle
+
+  if [ "$(printf "%s\n" "3.2" "$RUBY_VERSION" | sort -V | tail -n1)" = "$RUBY_VERSION" ]; then
+    bundle config set --local path 'vendor/bundle'
```
Comment (serggl) [inline ci/sunspot_test_script.sh:18]: @s4na can you explain the purpose of this conditional? would `bundle config set` + `bundle install` work the same way for all ruby versions?

Comment (serggl) [reply ci/sunspot_test_script.sh:18]: if this works the same way, then the conditional can be removed, right?

File: sunspot_rails/sunspot_rails.gemspec [hunk 2]
```diff
@@ -30,13 +30,15 @@ Gem::Specification.new do |s|
   s.executables   = `git ls-files -- bin/*`.split("\n").map{ |f| File.basename(f) }
   s.require_paths = ["lib"]
 
-  s.add_dependency 'rails', '>= 5'
+  s.add_dependency 'rails'
```
Comment (serggl) [inline sunspot_rails/sunspot_rails.gemspec:33]: why was the restriction removed removed?

**Review by serggl — COMMENTED** [review]
@s4na left few more questions. I see you removed development dependencies version constraints. Although I see test suite is passing, this might broke at some time in the future. I would vote to still pin gems to some explicit versions (latest at the moment)

**Comment by s4na** [comment]
@serggl 
Thank you for the review. I have made the necessary corrections.

File: sunspot/sunspot.gemspec [hunk 3]
```diff
@@ -25,12 +25,12 @@ Gem::Specification.new do |s|
   s.executables   = `git ls-files -- bin/*`.split("\n").map{ |f| File.basename(f) }
   s.require_paths = ["lib"]
 
-  s.add_dependency 'rsolr', '>= 1.1.1', '< 3'
+  s.add_dependency 'rsolr', '>= 1.1.1', '< 2.6'
```
Comment (s4na) [inline sunspot/sunspot.gemspec:28]: Since RSolr 2.6 is causing errors, it needs to be addressed in a separate PR.

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -2,25 +2,8 @@ name: CI
...
-          concurrent_skipping: 'same_content'
-          github_token: ${{ github.token }}
-          do_not_skip: '["pull_request"]'
-          cancel_others: 'true'
-          skip_after_successful_duplicate: 'false'
-          
+jobs:  
   build:
-    needs: skip_concurrent_build
-    if: ${{ needs.skip_concurrent_build.outputs.should_skip != 'true' }}
```
Comment (serggl) [inline .github/workflows/ci.yml:23]: @s4na would you mind bringing this back? I'm totally fine merging this afterwards

Comment (s4na) [reply .github/workflows/ci.yml]: OK

**Comment by s4na** [comment]
Thank you
